package-res/resources/web/pentaho/util/arg.js

```diff
@@ -20,23 +20,73 @@ define([
 
   var A_slice = Array.prototype.slice;
 
+  /**
+   * @name pentaho.util.arg
+   * @class
+   * @amd pentaho/util/arg
+   *
+   * @classDesc Util methods for keyword args.
+   *
+   * The AMD module returns the arg utils.
+   *
+   * @description Performs actions on keyword args.
+   */
   return {
+    /**
+     * Provides the optional property value of a given object or the given default value if the given object does
+     * not contain the sought after property value.
+     *
+     * @param {?object} [obj] The object to search for given propName param.
+     * @param {String} [propName] Property name to find in obj param.
+     * @param {any} [defaultValue] The default value to return if propName param is not found. The default value of this param is undefined and is considered specified when its value is not a nully value. {@link Nully}
+     *
```

Comment:
So, `[obj]` — optional — means you can omit the argument altogether, like calling it this way: `arg.optional()`. When one argument is optional, all arguments to its right should be optional as well.
And `{?object}` — nullable — means that the value of the argument may be null (we would actually want to mean _nully_ in this case, but whatever):
```js
var keyArgs = null; 
var value = arg.optional(keyArgs, "prop");
```
In this function, I'd say that the only optional argument is `defaultValue`.
Also, please do not use `{String}`, but use `{string}` instead. The same applies to the other "simple" primitive classes: `Number` and `Boolean`. Instances of String, Number and Boolean are objects, and quite different from their primitive, typeof == {string, number, boolean} counterparts.
For functions and objects, it's a bit more complicated.